### DISCUSSION TITLE
overriding default_path for each sfz mapping

### DISCUSSION THREAD

**Discussion by michael02022** [sfztools/sfizz#1231] [Requests]
Hi, I was testing a concept I had, which is very simple:

rhodes_map.sfz
```
<group>
<region> sample=rhodes_36.wav key=36
<region> sample=rhodes_37.wav key=37
<region> sample=rhodes_38.wav key=38
<region> sample=rhodes_39.wav key=39
<region> sample=rhodes_40.wav key=40
...
```

piano_map.sfz
```
<group>
<region> sample=piano_36.wav key=36
<region> sample=piano_37.wav key=37
<region> sample=piano_38.wav key=38
<region> sample=piano_39.wav key=39
<region> sample=piano_40.wav key=40
...
```
We have these sfz mappings, only contains `<group>` and `<region>` headers. As you can see, no path is given for these samples.

Now, with `#include` we can call these mappings with the correct path using the `<control>` header to each them, and under that header, there is a `default_path` opcode that goes under the called mapping. Each mapping has different `default_path` value. And it will work correctly in this way using absolute paths. (not tested with relative paths yet)

```
<control>
#define $USERPATH F:/My SFZs

<master>
ampeg_release=1

<control>
default_path=$USERPATH/Rhodes/Samples/
#include "$USERPATH/Rhodes/rhodes_map.sfz"

<master>
ampeg_release=3

<control>
default_path=$USERPATH/Grand/Samples/
#include "$USERPATH/Grand/piano_map.sfz"
```
Sforzando loads the samples without problem with this, but in Sfizz doesn't work.

**Comment by redtide** [comment]
Might be an issue, but since it started as a discussion, I'll leave it here. It can be moved later if necessary. @paulfd

**Comment by michael02022** [comment]
UPDATE: It turns out it works, but not with absolute paths.

**Comment by paulfd** [comment]
Thanks for the report! It does seem like something we should support.

**Reply by michael02022** [reply]
Thanks, I should mention again this works in sfizz if the `$USERPATH` variable is a relative path (like `$USERPATH ../..`) but good to know at least will verify it
